cadence/contracts/FlowYieldVaultsScheduler.cdc

```diff
@@ -177,50 +180,75 @@ access(all) contract FlowYieldVaultsScheduler {
+                    priority: priority,
+                    priorityRaw: priorityRaw,
+                    executionEffort: executionEffort,
+                    scanForStuck: scanForStuck
+                )
+            }
+        }
+
+        /// Self-reschedules the Supervisor for perpetual operation.
+        ///
+        /// This function handles the scheduling of the next Supervisor execution,
+        /// including fee estimation, withdrawal, and transaction scheduling.
+        ///
+        /// @param recurringInterval: The interval in seconds until the next execution
+        /// @param priority: The priority level for the scheduled transaction
+        /// @param priorityRaw: The raw priority value (UInt8) for data serialization
+        /// @param executionEffort: The execution effort estimate for the transaction
+        /// @param scanForStuck: Whether to scan for stuck yield vaults in the next execution
+        access(Schedule) fun scheduleNextRecurringExecution(
+            recurringInterval: UFix64,
+            priority: FlowTransactionScheduler.Priority,
+            priorityRaw: UInt8,
+            executionEffort: UInt64,
+            scanForStuck: Bool
+        ) {
+            let nextTimestamp = getCurrentBlock().timestamp + recurringInterval
+            let supervisorCap = FlowYieldVaultsSchedulerRegistry.getSupervisorCap()
+
+            if supervisorCap == nil || !supervisorCap!.check() {
+                return
+            }
+
+            let est = FlowYieldVaultsScheduler.estimateSchedulingCost(
+                timestamp: nextTimestamp,
+                priority: priority,
+                executionEffort: executionEffort
+            )
+            let baseFee = est.flowFee ?? FlowYieldVaultsScheduler.MIN_FEE_FALLBACK
+            let required = baseFee * FlowYieldVaultsScheduler.FEE_MARGIN_MULTIPLIER
+
+            if let vaultRef = self.feesCap.borrow() {
+                if vaultRef.balance >= required {
+                    let fees <- vaultRef.withdraw(amount: required) as! @FlowToken.Vault
+
+                    let nextData: {String: AnyStruct} = {
+                        "priority": priorityRaw,
+                        "executionEffort": executionEffort,
+                        "recurringInterval": recurringInterval,
+                        "scanForStuck": scanForStuck
                     }
+
+                    let selfTxn <- FlowTransactionScheduler.schedule(
+                        handlerCap: supervisorCap!,
+                        data: nextData,
+                        timestamp: nextTimestamp,
+                        priority: priority,
+                        executionEffort: executionEffort,
+                        fees: <-fees
+                    )
+
+                    emit SupervisorRescheduled(
+                        scheduledTransactionID: selfTxn.id,
+                        timestamp: nextTimestamp
+                    )
+
+                    destroy selfTxn
```

Comment:
Not tracking scheduled txns seems like an anti-pattern - we'd need to index offchain to get scheduled txn status versus the onchain status query - e.g. what we see in the AutoBalancers.

Options to address are limited now that we have a live deployment - can't add resource fields that would be helpful. We could store the resource in account storage via a contract-level method.

Maybe that's a matter for a v2 resource as I'd consider a non-blocker to get the Supervisor process back online.

Reply:
I think we can definitely still consider the "ideal" case for this, even if it requires a redeploy as this is somewhat a "supporting" contract so i feel like redeployments here are more acceptable

Reply:
Not sure I understand why the clean up has to be conditional on if it's internally or externally managed? why does that matter in this case? shouldn't we just check the IDs, and clean up if the tx ID matches and has been executed, and don't if it doesn't match and has not executed?